arch/arm/core/aarch64/reset.S

```diff
@@ -115,8 +115,31 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 	eret
```

Comment:
Uhm, not following here. `z_arch_el3_plat_init` is being called by the reset routine when we are still in EL3 and before dropping into EL1.

Reply:
>Uhm, not following here. `z_arch_el3_plat_init` is being called by the reset routine when we are still in EL3 and before dropping into EL1.

Correct, hence why this workaround works.

Sorry, let me rephrase a little:

Considering that `ICC_SRE_EL3` is an ARMv8-defined register that is mandatory by spec, and defined by spec to reset to disable all lower-level accesses to `ICC_SRE_EL*`, shouldn't this be part of architectural setup (i.e. the AArch64 arch setup in EL3, i.e. in the EL3 portion of `__start`) as opposed to something that every AArch64 platform that boots from EL3 will end up having to put in `z_arch_el3_plat_init`?

Or am I misunderstanding the distinction between "platform-specific setup" and "architecture-specific setup" here?

Reply:
`ICC_SRE_EL3` is initialized as part of a platform-specific setup because it is something related to the GIC while in the reset code we are keeping only the "core" stuff, of course we can argue if the GIC is more platform or architecture :)

Reply:
>because it is something related to the GIC while in the reset code we are keeping only the "core" stuff, of course we can argue if the GIC is more platform or architecture

Good point I suppose.

*****

Also, I took another look and it turns out I was incorrect with `ICC_SRE_EL3 is [...] mandatory by spec`. There's no mention of it in the description of `ICC_SRE_EL3`, but `ID_AA64PFR0_EL1.GIC` states "0000 No GIC system registers are supported", which rather implies it's optional.

*****

What is Zephyr's usual model for standard, but optional, architectural features that require differences in early boot?

Reply:
Moved to #32205 because not related to this PR.